projects/RabbitMQ.Client/client/events/AsyncEventingBasicConsumer.cs

```diff
@@ -86,5 +88,13 @@ public override async Task HandleChannelShutdown(object channel, ShutdownEventAr
                 await _shutdownWrapper.InvokeAsync(this, reason).ConfigureAwait(false);
             }
         }
+
+        private async Task BasicDeliverWrapper(BasicDeliverEventArgs eventArgs)
+        {
+            using (Activity activity = RabbitMQActivitySource.HasListeners ? RabbitMQActivitySource.Deliver(eventArgs) : default)
```

Comment:
There's a `HasListeners` check both here and in `Deliver` (for almost all activity initializations). Maybe not necessary to call it twice?

Reply:
hey @stebet  🙂 .. 
+1 for the traction  , can we expect anything on Nuget, maybe this side of the week ? or we are still a long way off

Reply:
Please do **not** ask for updates like this. This feature is set for version 7 which is still months away.